src/hooks/useMaxAmountIn.test.tsx

```diff
@@ -0,0 +1,17 @@
+import { TEST_ALLOWED_SLIPPAGE, TEST_TRADE_EXACT_INPUT } from 'test-utils/constants'
```

Comment:
Does `useMaxAmountIn` need to be a separate hook? It looks like it just wraps `trade.maximumAmountIn`.

Reply:
it was a chunk of code that i started re-using in multiple places in #6538 so i moved it here and wrote a test for it rather than copy-pasting